lib/puppet/file_serving/terminus_selector.rb

```diff
@@ -21,6 +26,8 @@ def select(request)
       else
         Puppet[:default_file_terminus]
       end
+   when "http","https"
+     :http
```

Comment:
No, this block here is plain necessary to enable `source` to use the correct indirector terminus.

In a perfect world, the other place would not have to inspect the URL at all. However, since the indirector key needs escaping (special handling for HTTP URLs), we need to work around that outside the indirector.